ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/participant_pruning_service.proto

```diff
@@ -50,3 +61,13 @@ message PruneRequest {
 message PruneResponse {
   // Empty for now, but may contain fields in the future
 }
+
+message GetOffsetByTimeRequest {
+  // Inclusive ``record_time`` timestamp up to and at which the ledger is to be pruned.
+  google.protobuf.Timestamp prune_up_to = 1;
+}
+
+message GetOffsetByTimeResponse {
+  // The absolute, inclusive offset of the last event/transaction at or before the ``prune_up_to`` input timestamp.
```

Comment:
@oliverse-da : what do you think about explaining the semantics of the endpoint here by relating a concrete request-response-pair to the observations possible on the completion streams and transaction streams for different parties; i.e., something along the lines that up to the returned offset all completions and transactions will have an `effective_at` field that is lesser or equal to the given timestamp.